fbw-common/src/systems/instruments/src/EFB/Apis/Navigraph/Components/Authentication.tsx

```diff
@@ -24,48 +25,44 @@ export type NavigraphAuthInfo =
     };
 
 export const useNavigraphAuthInfo = (): NavigraphAuthInfo => {
-  const navigraph = useNavigraph();
-
-  const [tokenAvail, setTokenAvail] = useState(false);
-  const [subscriptionStatus, setSubscriptionStatus] = useState<NavigraphSubscriptionStatus>(
-    NavigraphSubscriptionStatus.Unknown,
-  );
-  const [username] = usePersistentProperty('NAVIGRAPH_USERNAME');
-
-  useInterval(
-    () => {
-      if (tokenAvail !== navigraph.hasToken && navigraph.hasToken) {
-        navigraph
-          .fetchSubscriptionStatus()
-          .then(setSubscriptionStatus)
-          .catch(() => setSubscriptionStatus(NavigraphSubscriptionStatus.Unknown));
-      } else if (!navigraph.hasToken) {
-        setSubscriptionStatus(NavigraphSubscriptionStatus.None);
-      }
-
-      setTokenAvail(navigraph.hasToken);
-    },
-    1000,
-    { runOnStart: true },
-  );
-
-  if (tokenAvail) {
-    return { loggedIn: tokenAvail, username, subscriptionStatus };
+  const navigraphAuth = useNavigraphAuth();
+
+  if (navigraphAuth.user) {
+    return {
+      loggedIn: true,
+      username: navigraphAuth.user.preferred_username,
+      subscriptionStatus:
+        navigraphAuth.user.subscriptions.length >= 2
```

Comment:
This seems sus and prone to break... Surely they provide more useful data to determine what the user has rights to?

Reply:
They give you `fmsdata` and `charts` for an Ultimate user. Would you prefer checking for those two values?